### PR TITLE
Add initial canary workflow

### PR DESCRIPTION
**Pull request by jdisanti**
This adds a GitHub Action workflow to run the Lambda canary from smithy-rs every 24 hours.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/ci/run-canary.sh [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# LAMBDA_TEST_S3_BUCKET_NAME:
+#     The name of the S3 bucket for the canary Lambda to interact with
+# LAMBDA_EXECUTION_ROLE_ARN:
+#     The ARN of the role that the Lambda will execute as
+
+# TODO: Make the canary Cargo.toml generator output an empty workspace so that
+# the aws-sdk-rust root manifest doesn't cause the build to fail. Then remove this `rm`.
+rm -f "$(git rev-parse --show-toplevel)/Cargo.toml"
+
+set -e
```
Comment (rcoh) [inline tools/ci/run-canary.sh:30]: usually `set -e` is first

File: tools/ci/run-canary.sh [hunk 2]
```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
```
Comment (rcoh) [inline tools/ci/run-canary.sh:1]: can you enable the shellcheck intellij plugin? it will point out a few things to tweak

Comment (rcoh) [reply tools/ci/run-canary.sh:1]: but TBH, I would go full meta and write this with the Rust SDK :-)

**Review by rcoh — APPROVED** [review]
LGTM, but the canaries are all failing with invalid S3 metadata 😂

File: tools/ci/canary-matrix.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import subprocess
+import sys
+import unittest
+
+
+def get_sdk_versions(desired_versions_count, aws_config_metadata_lines):
+    versions = []
+    for line in aws_config_metadata_lines:
+        parsed = json.loads(line)
+        if parsed["yanked"] != True:
```
Comment (rcoh) [inline tools/ci/canary-matrix.py:21]: this method would be easier to test if it didn't expect JSON as strings

File: tools/ci/run-canary.sh (hunk 2 above)
Comment (jdisanti) [reply tools/ci/run-canary.sh]: It's actually a pretty good exercise of the Rust SDK in addition to the canary Lambda, so I'm onboard with this.

**Pull request merged**